File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/configuration/EnumConverterConfiguration.java
package org.openapitools.configuration;

import org.openapitools.model.FruitType;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.convert.converter.Converter;

@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
public class EnumConverterConfiguration {

    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.fruitTypeConverter")
    Converter<String, FruitType> fruitTypeConverter() {
        return new Converter<String, FruitType>() {
            @Override
            public FruitType convert(String source) {
                return FruitType.fromValue(source);
            }
        };
    }

}

File: samples/server/petstore/kotlin-springboot-integer-enum/src/main/kotlin/org/openapitools/configuration/EnumConverterConfiguration.kt
package org.openapitools.configuration

import org.openapitools.model.ReasonCode

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.convert.converter.Converter

@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
class EnumConverterConfiguration {

    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.reasonCodeConverter"])
    fun reasonCodeConverter(): Converter<kotlin.Int, ReasonCode> {
        return object: Converter<kotlin.Int, ReasonCode> {
            override fun convert(source: kotlin.Int): ReasonCode = ReasonCode.forValue(source)
        }
    }

}
